src/golang/lib/workflow/operator/base.go

```diff
@@ -204,15 +188,68 @@ func (bo *baseOperator) InitializeResult(ctx context.Context, dagResultID uuid.U
 		ctx,
 		dagResultID,
 		bo.ID(),
+		&bo.execState,
 		bo.db,
 	)
 	if err != nil {
 		return errors.Wrap(err, "Failed to create operator result record.")
 	}
+
 	bo.resultID = operatorResult.Id
+
 	return nil
 }
 
+func (bo *baseOperator) Poll(ctx context.Context) (*shared.ExecutionState, error) {
+	if bo.jobName == "" {
+		return nil, errors.Newf("Internal error: a job name was not set for this operator.")
+	}
+
+	// The operator is already terminated. No need to update status.
+	if bo.execState.Terminated() {
+		return bo.ExecState(), nil
+	}
+
+	status, err := bo.jobManager.Poll(ctx, bo.jobName)
+	if err != nil {
+		// If the job does not exist, this could mean that
+		// 1) it is hasn't been run yet (pending),
+		// 2) it has run already at sometime in the past, but has been garbage collected
+		// 3) it has run already at sometime in the past, but did not go through the job manager.
+		//    (this can happen when the output artifacts have already been cached).
+		if err == job.ErrJobNotExist {
```

Comment:
@hsubbaraj-spiral I think this is something we can reuse for lambda and other async executor. I'm considering making an error type like `ErrJobIsAsync` and handle it here. Other parts seems to be reusable